### PR TITLE
Added dcumentation for ecm and qs functions

### PR DESCRIPTION
**Pull request by mayankray2020**
#### References to other Issues or PRs
Taking over #19971 

* ntheory
  * Added documentation of ecm and qs functions

### CODE REVIEW

File: doc/src/modules/ntheory.rst [hunk 1]
```diff
@@ -192,3 +192,56 @@ Ntheory Functions Reference
...
+===========
+
+The `qs` function is a subexponential factoring algorithm, the fastest
+factoring algorithm for numbers within 100 digits. The time complexity of
+`qs` is dependent on the size of the number so it is used if the number contains
+large factors. Due to this while factoring numbers first `ecm` is used to get
+smaller factors of around ~15digits then `qs` is used to get larger factors.
+
+For factoring `2709077133180915240135586837960864768806330782747` which is a semi-prime number
+with two 25 digit factors. `qs` is able to factorize this in    around 248s.
```
Comment (abh2k) [inline doc/src/modules/ntheory.rst:238]:
```suggestion
with two 25 digit factors. `qs` is able to factorize this in around 248s.
```

**Comment by mayankray2020** [comment]
@abh2k @czgdp1807

**Comment by abh2k** [comment]
@Yathartha22  Please review this.

File: doc/src/modules/ntheory.rst [hunk 2]
```diff
@@ -192,3 +192,56 @@ Ntheory Functions Reference
...
+.. module:: sympy.ntheory.qs
+
+QS function
+===========
+
+The `qs` function is a subexponential factoring algorithm, the fastest
+factoring algorithm for numbers within 100 digits. The time complexity of
+`qs` is dependent on the size of the number so it is used if the number contains
+large factors. Due to this while factoring numbers first `ecm` is used to get
+smaller factors of around ~15digits then `qs` is used to get larger factors.
```
Comment (abh2k) [inline doc/src/modules/ntheory.rst:235]:
```suggestion
smaller factors of around ~15 digits then `qs` is used to get larger factors.
```

File: doc/src/modules/ntheory.rst [hunk 3]
```diff
@@ -192,3 +192,56 @@ Ntheory Functions Reference
...
 .. autofunction:: pi_hex_digits
+
+.. module:: sympy.ntheory.ecm
+
+ECM function
+============
+
+The `ecm` function is a subexponential factoring algorithm capable of factoring
+numbers of around ~35 digits comfortably within few seconds. The time complexity
+of `ecm` is dependent on the smalled proper factor of the number. So even if the
```
Comment (abh2k) [inline doc/src/modules/ntheory.rst:203]:
```suggestion
of `ecm` is dependent on the smallesst proper factor of the number. So even if the
```

**Comment by abh2k** [comment]
I have given a minor  suggestion.  Apart from this it seem good.

**Comment by mayankray2020** [comment]
@czgdp1807 Can we merge this??

**Comment by mayankray2020** [comment]
@czgdp1807 can this be merged??

**Pull request merged**